census/estimator/trainer/task.py

```diff
@@ -0,0 +1,162 @@
+import argparse
+import json
+import os
+
+import model
+
+from tensorflow.contrib.learn.python.learn import learn_runner
+from tensorflow.contrib.learn.python.learn.utils import (
+    saved_model_export_utils)
+from tensorflow.contrib import metrics, learn
+
+
+def generate_experiment_fn(train_file,
+                           eval_file,
+                           num_epochs=None,
+                           train_batch_size=40,
+                           eval_batch_size=40,
+                           embedding_size=8,
+                           serving_input_fn_type='JSON',
+                           hidden_units=None,
+                           **experiment_args):
+  def _experiment_fn(output_dir):
+    train_input = model.generate_input_fn(
+        train_file, num_epochs=num_epochs, batch_size=train_batch_size)
+    eval_input = model.generate_input_fn(
+        eval_file,
+        batch_size=eval_batch_size,
+        skip_header_lines=1,
+        shuffle=False
+    )
+    if args.serving_input_fn_type == 'JSON':
```

Comment:
@brandondutra Interesting. I still believe it's in the user's best interest to show them how to use JSON rather than CSV. CSV just doesn't make sense for the use cases that online prediction is targeting.